pkg/vault/contracts/RouterCommon.sol

```diff
@@ -264,31 +263,12 @@ abstract contract RouterCommon is IRouterCommon, VaultGuard, ReentrancyGuardTran
         signatureParts.v = v;
     }
 
-    /**
-     * @dev Returns excess ETH back to the contract caller. Checks for sufficient ETH balance are made right before
-     * each deposit, ensuring it will revert with a friendly custom error. If there is any balance remaining when
-     * `_returnEth` is called, return it to the sender.
-     *
-     * Because the caller might not know exactly how much ETH a Vault action will require, they may send extra.
-     * Note that this excess value is returned *to the contract caller* (msg.sender). If caller and e.g. swap sender
-     * are not the same (because the caller is a relayer for the sender), then it is up to the caller to manage this
-     * returned ETH.
-     */
     function _returnEth(address sender) internal {
```

Comment:
I think I'd leave this one out, or maybe put the whole function in another library? (that'd be out of scope for this PR).

This function doesn't really use WETH, and as the deleted comment states it's cheaper to do it this way than reading transient storage unconditionally. That's why I'm not super convinced.

It is true that we will likely need this elsewhere though. But maybe instead of WETH we can do it with the `isReturnEthLockedSlot` (weird, but it might work and we wouldn't need to actually read the slot)

Reply:
Makes sense, I'll remove the refactor of the function `_returnEth` from this PR.